### ISSUE TITLE
Features, and Roadmap

### ISSUE THREAD

**Issue by alireza-hariri** [ali77gh/bake-rs#1]
I am interested in this project.
Unfortunately, my experience with rust is quite limited.

I can work on some features but it may takes time and patient.
do you have some good first issues?

Anyway Here is some suggestions for future features:
1- TUI is garte but I could not see much value in GUI (maybe I couldn't see).
2- There is some other make alternatives that can help a lot in finding useful features (`just`, `Taskfile`, `Invoke`)
3- reading `.env` files 
4- running commands in parallel
5- helper for LD_LIBRARY_PATH modification (and other envirement helpers)

let me know how do you tink about this suggestions

**Comment by ali77gh** [comment]
Hi🖐️.
Cool!

GUI can be really useful. 

let's say you can write a simple yaml for your grandpa to backup his file. and It can be run with a simple click.
or
I can write a simple yaml file for my non-programmer co-worker (in my case electronic engineer) and she/he can compile codebase with specified envs and upload code to micro-controllers that is already done throw arduino-cli which is kinda hard for non-programmers. 

But I think we should make a different binary for GUI like -> 'bake' for cli and tui and 'bake-gui' for GUI

What do you think about that?

pyInvoke is so flexible cause it's using a programming language. I was thinking about embedding Lua into bake to do this.

What do you think about this?

Taskfile is so cool because it using [sh](https://github.com/mvdan/sh) and its same script for linux/mac/windows. but it's reinventing wheel by implementing 'if' and 'for' and I think I can make a better script runner.

I agree with reading .env file. Nice!

Parallel is cool. But I never have no Idea how yaml should be and how cli/tui/gui should be. share your ideas.

Thanks for your time and attention.

My last Question is: Can this project be a successor for 'just', taskfile or invoke? what is your honest answer?

**Comment by alireza-hariri** [comment]
oh, separate bins for gui and tui make sense to me.

I didn't know taskfile could support `sh` cool.

Lua is a good choice but it has some weird things. (arrays start with 1). 
I liked [rahi](https://github.com/rhaiscript/rhai) scripting language syntax check it.

Predicting the future popularity of a package is very hard and may depend on non-technical things.
but most open-source projects don't get the chance. overfocusing on the results could be misleading and demotivating here.
 comparing small projects with widely adopted ones is somehow bad. Even the successful taskfile should not be compared with the make.
every project needs some early adopters to continue. but this is too soon for wider adaptation.

**Comment by ali77gh** [comment]
Rahi is cool but Lua is known by people.

I agree about what you think. But this time I fill like it's not just a fun project for myself. And if I  want this project to be used by others I should be much better than these tools. and being better is not about implementation. It's about Ideas and Design Decisions.

So when I think about our Ideas and I fill like It can be a better alternative, I get motivated no matter what happens I will finish this project and also pay attention to that 'non-technical' things.

I made a ['Project' on my repo](https://github.com/users/ali77gh/projects/5/views/2) to share my ideas and planning about this project publicly.

It's my first time using this feature on github so I don't know best practices.

I can Add you to this board if you want.

**Comment by alireza-hariri** [comment]
cool,
I will follow your news.

I agree about non-technicals.
for example, I was using `pyInvoke`, and simple issues like removing the color from the command output were a real problem.
